batchflow/tests/model_save_load_test.py

```diff
@@ -77,59 +74,85 @@ def test_run(self, save_path, pipelines, model_class):
 
         Predictions from save_pipeline and from load_pipeline should be equal
         """
-        save_pipeline, load_pipeline = pipelines(model_class)
-
         save_tmpl = (Pipeline()
                      .save_model('model', path=save_path + I("current").str())
-                     .train_model('model', B('images'), B('labels'), fetches='loss'))
-
-        save_pipeline = save_pipeline + save_tmpl
-        save_pipeline.run(BATCH_SIZE, n_epochs=1, bar=True)
-        saved_predictions = save_pipeline.get_variable('predictions')
+                    )
 
         load_tmpl = (Pipeline()
-                     .load_model('dynamic', C('model_class'), 'model', path=save_path + I("current").str()))
+                     .load_model('model', C('model_class'), 'dynamic', path=save_path + I("current").str())
+                    )
+
+        train_pipeline, predict_pipeline = pipelines(model_class)
 
-        load_pipeline = load_tmpl + load_pipeline
-        load_pipeline.run(BATCH_SIZE, n_epochs=1, bar=True)
-        loaded_predictions = load_pipeline.get_variable('predictions')
+        train_pipeline = train_pipeline + save_tmpl
+        train_pipeline.run(BATCH_SIZE, n_epochs=1)
+        saved_predictions = train_pipeline.v('predictions')
+
+        predict_pipeline = load_tmpl + predict_pipeline
+        predict_pipeline.run(BATCH_SIZE, n_epochs=1)
+        loaded_predictions = predict_pipeline.v('predictions')
 
         assert (np.concatenate(saved_predictions) == np.concatenate(loaded_predictions)).all()
 
     def test_now(self, save_path, pipelines, model_class):
         """
         Test model loading and saving with `save_model_now`  and `load_model_now`
         """
-        save_pipeline, load_pipeline = pipelines(model_class)
+        train_pipeline, predict_pipeline = pipelines(model_class)
 
-        save_pipeline.run(BATCH_SIZE, n_epochs=1)
-        saved_predictions = save_pipeline.get_variable('predictions')
-        save_pipeline.save_model_now('model', path=save_path)
+        train_pipeline.run(BATCH_SIZE, n_epochs=1)
```

Comment:
Consider using `next_batch`, because you compare only the predictions from the last batch anyway